### PR TITLE
Docs improvements

### PR DESCRIPTION
**Pull request by Grimmys**
According to #2618, this PR makes the following changes:

- Remove historical `docs/howto_release_pygame.txt` file
- Fixed start up of local documentation via `docs/__main__.py`
- Add the optional flag `--fullgeneration` (alias `--f`) to the `docs` command to be able to fully rebuild the documentation
- Create a first version for the docs README

Regarding the optional flag, I added a new entry in `user_options` of the docs command, and a new attribute directly linked to this option, following the inner workings of setuptools/distutils.  
If the flag is present, `buildconfig/makeref.py` will be called with the arg `full_generation`, and the `-E` option will be added to sphinx-build command.
To sum up:
`python setup.py docs` -> unchanged behavior
`python setup.py docs --fullgeneration` -> call sphinx-build command with `-E` as an additional option

About the README, it's really a first draft.
Many other things should be added, I have not so much experience with the docs package and [Starbuck said he has some ideas to add](https://github.com/pygame/pygame/issues/2618#issuecomment-854572757).
Moreover, I'm not a native: it would be really great if a native takes the time to verify the grammar and to reformulate the weird sentences!

### CODE REVIEW

**Comment by Starbuck5** [comment]
Sorry I haven't gotten to this PR in so long.

I like the changes!

The organization of the README.md strikes me as a little bit convoluted, I think changing the first main header to something like "Generating the docs" would help. Just since there is an "Access documentation locally" and then an "Access to documentation" header.

The only English mistake I found was "advice" should be "advise" in the contribution section.

But I think the contribution section should be changed:
- It needs a message encouraging people to contribute to the docs
- I think the "check out the sphinx docs" request is intimidating to newcomers. Maybe it could be phrased in a more informational way... like "pygame uses Sphinx to generate html documentation from restructured text (rst) files. Sphinx docs website is a good resource".

File: docs/README.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
+## Access documentation locally
```
Comment (illume) [inline docs/README.md:1]: There are two audiences here:
- 1) people looking to contribute/develop/write docs
- 2) people trying to read pygame docs.

Doesn't have to be in this PR, but we probably also need something for people looking into the docs folder looking for pygame docs.

**Comment by illume** [comment]
Note we have: https://www.pygame.org/wiki/Hacking

There's a docs section in there - feel free to take from that. Probably when this README is complete for contributing to docs, then we should link to it from the Hacking page (DRY) so we only have one to maintain.

File: docs/__main__.py [hunk 2]
```diff
@@ -2,21 +2,24 @@
...
     path, last = os.path.split(path)
     if last:
         for p in iterpath(path):
             yield p
         yield last
 
+
 pkg_dir = os.path.dirname(os.path.abspath(__file__))
-main_page = os.path.join(pkg_dir, 'index.html')
+main_page = os.path.join(pkg_dir, 'generated/index.html')
```
Comment (illume) [inline docs/__main__.py:22]: oops. Good catch.

**Review by illume — APPROVED** [review]
👍 thanks.

I personally think we should merge this and improve it in separate PRs.

**Review by Starbuck5 — CHANGES_REQUESTED** [review]
docs/README.md isn't merge ready, in my opinion,

@Grimmys you could take it out of this PR and then we can push everything else through, or stay the course and do it all as one PR.

**Comment by Grimmys** [comment]
Thanks for your reviews! 

I don't know if I should split this PR... 
In my opinion, I should fix the grammar and formulation issues, and it will be ready to be merge. 

The README doesn't have to be complete (everyone can have things to add) for this PR, but should be at least in a minimal "valid shape".

**Comment by Grimmys** [comment]
I removed the commits about the README from this branch/PR, we should now be able to merge it.

File: docs/__main__.py (hunk 2 above)
Comment (ankith26) [reply docs/__main__.py:22]: minor code nitpick here, `os.path.join(pkg_dir, 'generated/index.html')` can also be `os.path.join(pkg_dir, 'generated', 'index.html')`

**Review by ankith26 — APPROVED** [review]
This PR LGTM! 🎈 🥳 
After the merge conflict is fixed, this PR should be good to go

File: docs/__main__.py (hunk 2 above)
Comment (Grimmys) [reply docs/__main__.py:22]: oh yeah you are right, definitively better to rely on the default directory separator provided by `os.path.join`, thanks!

**Review by Starbuck5 — APPROVED** [review]
Let's get this merged!

Just have to wait for the tests to run on the merge.
